internal/provider/resource_cloudflare_api_shield.go

```diff
@@ -0,0 +1,117 @@
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/pkg/errors"
+)
+
+func resourceCloudflareAPIShield() *schema.Resource {
+	return &schema.Resource{
+		Schema:        resourceCloudflareAPIShieldSchema(),
+		CreateContext: resourceCloudflareAPIShieldCreate,
+		ReadContext:   resourceCloudflareAPIShieldRead,
+		UpdateContext: resourceCloudflareAPIShieldUpdate,
+		DeleteContext: resourceCloudflareAPIShieldDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: nil,
+		},
+		Description: heredoc.Doc(`
+			Provides a resource to manage API Shield configurations.
+		`),
+	}
+}
+
+func resourceCloudflareAPIShieldCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*cloudflare.API)
+	zoneID := d.Get("zone_id").(string)
+
+	as, err := buildAPIShieldConfiguration(d)
+	if err != nil {
+		return diag.FromErr(errors.Wrap(err, fmt.Sprintf("failed to create API Shield Configuration")))
+	}
+
+	_, err = client.UpdateAPIShieldConfiguration(ctx, cloudflare.ZoneIdentifier(zoneID), cloudflare.UpdateAPIShieldParams{AuthIdCharacteristics: as.AuthIdCharacteristics})
+	if err != nil {
+		return diag.FromErr(errors.Wrap(err, fmt.Sprintf("failed to create API Shield Configuration")))
+	}
+
+	return resourceCloudflareAPIShieldRead(ctx, d, meta)
+}
+
+func resourceCloudflareAPIShieldRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*cloudflare.API)
+	zoneID := d.Get("zone_id").(string)
+
+	as, _, err := client.GetAPIShieldConfiguration(ctx, cloudflare.ZoneIdentifier(zoneID))
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("failed to fetch API Shield Configuration: %w", err))
+	}
+
+	d.Set("auth_id_characteristics", flattenAPIShieldConfiguration(as.AuthIdCharacteristics))
+	d.Set("zone_id", zoneID)
+	d.SetId(zoneID)
+
+	return nil
+}
+
+func resourceCloudflareAPIShieldUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*cloudflare.API)
+	zoneID := d.Get("zone_id")
+
+	as, err := buildAPIShieldConfiguration(d)
+	if err != nil {
+		return diag.FromErr(errors.Wrap(err, fmt.Sprintf("failed to create API Shield Configuration")))
+	}
+
+	_, err = client.UpdateAPIShieldConfiguration(ctx, cloudflare.ZoneIdentifier(zoneID.(string)), cloudflare.UpdateAPIShieldParams{AuthIdCharacteristics: as.AuthIdCharacteristics})
+	if err != nil {
+		return diag.FromErr(errors.Wrap(err, fmt.Sprintf("failed to create API Shield Configuration")))
+	}
+
+	return resourceCloudflareAPIShieldRead(ctx, d, meta)
+}
+
+func resourceCloudflareAPIShieldDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*cloudflare.API)
+	zoneID := d.Get("zone_id")
+
+	_, err := client.UpdateAPIShieldConfiguration(ctx, cloudflare.ZoneIdentifier(zoneID.(string)), cloudflare.UpdateAPIShieldParams{nil})
```

Comment:
this doesn't seem to be working. the payload is `{"auth_id_characteristics":null}` however, i'm unsure what it should be. could you find out? we may need to update the Go library if it is expecting no keys too.

Reply:
I followed up with the team:
```
we usually tend to preserve configurations and not delete them.
The endpoint currently has only one configuration value. To remove that one can send as the payload using PUT:
{"auth_id_characteristics":[]}
``` 
I will update the PR to send an empty `auth_id_characteristics`